packages/frontend-2/components/server-management/SettingsDialog.vue

```diff
@@ -0,0 +1,185 @@
+import { LayoutDialog, FormTextInput, FormTextArea } from '@speckle/ui-components'
+import { useLogger } from '~~/composables/logging'
+import { convertThrowIntoFetchResult } from '~~/lib/common/helpers/graphql'
+import { serverInfoQuery } from '~~/lib/server-management/graphql/queries'
+import { serverInfoUpdateMutation } from '~~/lib/server-management/graphql/mutations'
+import type { Modifier, Reference } from '@apollo/client/cache'
+
+type FormValues = {
+  name: string
+  description: string
+  company: string
+  adminContact: string
+  termsOfService: string
+  inviteOnly: boolean
+}
+
+type ServerInfoUpdateVariables = {
+  info: FormValues
+}
+
+const props = defineProps<{
+  open: boolean
+}>()
+
+const emit = defineEmits<{
+  (e: 'update:open', val: boolean): void
+  (e: 'server-info-updated'): void
+}>()
+
+const logger = useLogger()
+const { triggerNotification } = useGlobalToast()
+const { handleSubmit } = useForm<FormValues>()
+const { result } = useQuery(serverInfoQuery)
+const { mutate: updateServerInfo } = useMutation(serverInfoUpdateMutation)
+
+const name = ref('')
+const description = ref('')
+const company = ref('')
+const adminContact = ref('')
+const termsOfService = ref('')
+const inviteOnly = ref<true | undefined>(undefined)
+
+const isOpen = computed({
+  get: () => props.open,
+  set: (newVal) => emit('update:open', newVal)
+})
+
+const dialogButtons = computed(() => [
+  {
+    text: 'Cancel',
+    props: { color: 'secondary', fullWidth: true, outline: true },
+    onClick: () => (isOpen.value = false)
+  },
+  {
+    text: 'Save',
+    props: { color: 'primary', fullWidth: true, outline: false },
+    onClick: onSubmit
+  }
+])
+
+const requiredRule = [isRequired]
+
+const updateServerInfoAndCache = async (variables: ServerInfoUpdateVariables) => {
+  await updateServerInfo(variables, {
+    update: (cache, result) => {
+      if (result?.data?.serverInfoUpdate) {
+        cache.modify({
+          fields: {
+            serverInfo: ((existingServerInfo: FormValues): FormValues => {
+              return {
+                ...existingServerInfo,
+                ...variables.info
+              }
+            }) as Modifier<FormValues | Reference>
+          }
+        })
+      }
+    }
+  }).catch(convertThrowIntoFetchResult)
+}
+
+const onSubmit = handleSubmit(async () => {
+  try {
+    await updateServerInfoAndCache({
+      info: {
+        name: name.value,
+        description: description.value,
+        company: company.value,
+        adminContact: adminContact.value,
+        termsOfService: termsOfService.value,
+        inviteOnly: inviteOnly.value || false
+      }
+    })
+    triggerNotification({
+      type: ToastNotificationType.Success,
+      title: 'Successfully saved',
+      description: 'Your server settings have been saved.'
+    })
+    isOpen.value = false
+    emit('server-info-updated')
+  } catch (error) {
+    logger.error(error)
```

Comment:
This is unreachable code. In `updateServerInfoAndCache` you're suppressing error throws with `convertThrowIntoFetchResult`, but you're also not returning the results from that function so there's no way for you to check if the operation succeeded or not.